include/directx/d3dx12.h

```diff
@@ -4037,6 +4037,660 @@ class CD3DX12_NODE_MASK_SUBOBJECT
+    FEATURE_SUPPORT_GET_NODE_INDEXED(BOOL, m_dArchitecture1, UMA);
+    FEATURE_SUPPORT_GET_NODE_INDEXED(BOOL, m_dArchitecture1, CacheCoherentUMA);
+    FEATURE_SUPPORT_GET_NODE_INDEXED(BOOL, m_dArchitecture1, IsolatedMMU);
+
+    // 18: D3D12 Options2
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions2, DepthBoundsTestSupported);
+    FEATURE_SUPPORT_GET(D3D12_PROGRAMMABLE_SAMPLE_POSITIONS_TIER, m_dOptions2, ProgrammableSamplePositionsTier);
+
+    // 19: Shader Cache
+    FEATURE_SUPPORT_GET_NAME(D3D12_SHADER_CACHE_SUPPORT_FLAGS, m_dShaderCache, SupportFlags, ShaderCacheSupportFlags);
+
+    // 20: Command Queue Priority
+    BOOL CommandQueuePrioritySupported(D3D12_COMMAND_LIST_TYPE CommandListType, UINT Priority) 
+    {
+        m_dCommandQueuePriority.CommandListType = CommandListType;
+        m_dCommandQueuePriority.Priority = Priority;
+
+        // TODO: Report invalid arguments?
+        if (FAILED(m_pDevice->CheckFeatureSupport(D3D12_FEATURE_COMMAND_QUEUE_PRIORITY, &m_dCommandQueuePriority, sizeof(D3D12_FEATURE_DATA_COMMAND_QUEUE_PRIORITY)))) {
+            return false;
+        }
+
+        return m_dCommandQueuePriority.PriorityForTypeIsSupported;
+    }
+
+    // 21: D3D12 Options3
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions3, CopyQueueTimestampQueriesSupported);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions3, CastingFullyTypedFormatSupported);
+    FEATURE_SUPPORT_GET(D3D12_COMMAND_LIST_SUPPORT_FLAGS, m_dOptions3, WriteBufferImmediateSupportFlags);
+    FEATURE_SUPPORT_GET(D3D12_VIEW_INSTANCING_TIER, m_dOptions3, ViewInstancingTier);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions3, BarycentricsSupported);
+
+    // 22: Existing Heaps
+    FEATURE_SUPPORT_GET_NAME(BOOL, m_dExistingHeaps, Supported, ExistingHeapsSupported);
+
+    // 23: D3D12 Options4
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions4, MSAA64KBAlignedTextureSupported);
+    FEATURE_SUPPORT_GET(D3D12_SHARED_RESOURCE_COMPATIBILITY_TIER, m_dOptions4, SharedResourceCompatibilityTier);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions4, Native16BitShaderOpsSupported);
+
+    // 24: Serialization
+    FEATURE_SUPPORT_GET_NODE_INDEXED(D3D12_HEAP_SERIALIZATION_TIER, m_dSerialization, HeapSerializationTier);
+
+    // 25: Cross Node
+    // FEATURE_SUPPORT_GET_NAME(D3D12_CROSS_NODE_SHARING_TIER, m_dCrossNode, SharingTier, CrossNodeSharingTier); // Duplicate in D3D12Options
+    FEATURE_SUPPORT_GET_NAME(BOOL, m_dCrossNode, AtomicShaderInstructions, CrossNodeAtomicShaderInstructions);
+    
+    // 27: D3D12 Options5
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions5, SRVOnlyTiledResourceTier3);
+    FEATURE_SUPPORT_GET(D3D12_RENDER_PASS_TIER, m_dOptions5, RenderPassesTier);
+    FEATURE_SUPPORT_GET(D3D12_RAYTRACING_TIER, m_dOptions5, RaytracingTier);
+
+    // 28: Displayable
+    FEATURE_SUPPORT_GET(BOOL, m_dDisplayable, DisplayableTexture);
+    FEATURE_SUPPORT_GET_NAME(D3D12_SHARED_RESOURCE_COMPATIBILITY_TIER, m_dDisplayable, SharedResourceCompatibilityTier, DisplayableSharedResourceCompatibilityTier);

```

Comment:
There is another `SharedResourceCompatibilityTier` cap in D3D12Options4. I checked the code in Device.cpp and found that while the code to check for these two caps are slightly different, the `SharedResourceCompatibilityTier` from `Displayable` is actuallyreading from `m_ValidationInfo.m_Options4.SharedResourceCompatibilityTier`. Does it suggest the two are essentially the same?

Reply:
That's a good question. From what I can tell, the Displayable field used to be experimental, and capable of returning values that Option4 would not. However, now the Displayable field isn't experimental any more, so it would appear to be a redundant cap. @alexgong14 might know more about it.

Aside: in the API there's caps that are definitely redundant, and I have trouble imagining that changing: options.MaxGPUVirtualAddressBitsPerResource and GpuVirtualAddressSupport.MaxGPUVirtualAddressBitsPerResource.

Reply:
The short answer is Yes.

Displayable surface is a D3D11 feature and we haven't yet allowed the use of it in D3D12. If you call D3D12 CheckFeatureSupport, the field DisplayableTexture will always be False. This indicates it is not supported by D3D12. For SharedResourceTier, both D3D11 Displayable and D3D12 Displayable share it with other D3D11/D3D12Options, the Output below is used by both D3D11_FEATURE_D3D11_OPTIONS5 and D3D11_FEATURE_DISPLAYABLE:

D3D11_SHARED_RESOURCE_TIER Output = D3D11_SHARED_RESOURCE_TIER_0;
            if (!IsExplicitReferenceDriver())
            {
                Output = D3D11_SHARED_RESOURCE_TIER
                    ( D3D11_SHARED_RESOURCE_TIER_0 + CD3D11FormatHelper::MediaSharedResourceTier
                        ( GetExtendedFormatFeatures()
                        , true ) // Feature_D3D1XDisplayable
                    );
            }
            
In my opinion, the reason Displayable includes SharedResourceTier is just for convenience, because Displayable requires current WDDM version to be no less than a certain WDDM Version, for D3D11 it is WDDM 2.7. So when people try to use Displayable but they get DisplayableTexture = false, they can also look at the SharedResourceTier to see if Displayable is not supported because their device does not meet the minimum requirement to support Displayable.

Reply:
Got it. Thanks for the explanation! If these two caps are the same thing, then I no longer need the `DisplayableSharedResourceTier` feature any more. I will remove this redundant API in the next commit.

Reply:
Resolved. The duplicate getter is removed.